eval/src/vespa/eval/tensor/sparse/sparse_tensor.cpp

```diff
@@ -29,10 +29,11 @@ using Cells = SparseTensor::Cells;
 void
 copyCells(Cells &cells, const Cells &cells_in, Stash &stash)
 {
-    for (const auto &cell : cells_in) {
+    cells = cells_in;
+    for (auto &cell : cells) {
         SparseTensorAddressRef oldRef = cell.first;
         SparseTensorAddressRef newRef(oldRef, stash);
-        cells[newRef] = cell.second;
+        cell.first = newRef;
```

Comment:
1 - A little comment about this not changing the hash value would be good.
2 - I guess you can can also write it as 
cell.first = SparseTensorAddressRef(cell.first, stash);
You should see what effect it has on speed. The compiler might find it easier or harder. 
noexcept on the SparseTensorAddressRef constructors might also make it easier on the optimizer.

Reply:
added some comments.
doing 2) or adding noexcept didn't have any impact on the produced code, so I left it as-is.